straxen/plugins/nveto_pulse_processing.py

```diff
@@ -84,12 +85,35 @@ def clean_up_empty_records(records, only_last=True):
     for ind, r in enumerate(records):
         if only_last:
             m_last_fragment = (r['record_i'] > 0) and (r['length'] < len(r['data']))
```

Comment:
Perhaps this is easier with the record_links:
```python
if only_last:
    if record_links[0][ind] == strax.NO_RECORD_LINK or record_links[1][ind] != strax.NO_RECORD_LINK:
        # we are not the last record as we don't have a link on the left (i.e. are first) or have a link on the right
        indicies_to_keep[n_indicies] = ind
        n_indicies += 1
```
        continue

Reply:
Nice, idea in the beginning I did not use record_links. But you are right since we are using them anyhow I can change it.